modules/local/Decoding/main.nf

```diff
@@ -0,0 +1,68 @@
+//decode Using 2 gpus and high resolution array
+process Decode_batch {
+    tag "$meta.id" 
+    label "gpu_process"
+    conda "${moduleDir}/environment.yml"
+    container 'docker://pytorch/pytorch:2.7.1-cuda11.8-cudnn9-runtime'
```

Comment:
Preferably these containers should be created with [Seqera Containers](https://seqera.io/containers/) and both Docker and Singularity containers should be present. (Both can be created with Seqera Containers). Take a look at other pipelines to have an example on how this should be implemented